log4j-core/src/main/java/org/apache/logging/log4j/core/config/HttpWatcher.java

```diff
@@ -103,34 +102,47 @@ private boolean refreshConfiguration() {
         try {
             final LastModifiedSource source = new LastModifiedSource(url.toURI(), lastModifiedMillis);
             final HttpInputStreamUtil.Result result = HttpInputStreamUtil.getInputStream(source, authorizationProvider);
+            // Update lastModifiedMillis
+            // https://github.com/apache/logging-log4j2/issues/2937
+            lastModifiedMillis = source.getLastModified();
```

Comment:
One bug was here: `lastModifiedMillis` didn't propagate from `source` the the `HttpWatcher`...